app/controllers/spree/amazon_controller.rb

```diff
@@ -88,18 +88,26 @@ def complete
 
   private
 
+  def gateway
+    @gateway ||= Spree::Gateway::Amazon.for_currency(current_order.currency)
+  end
+
+  def amazon_order
+    @amazon_order ||= SpreeAmazon::Order.new(
+      reference_id: current_order.amazon_order_reference_id,
+      total: current_order.total,
+      currency: current_order.currency,
+      gateway: gateway,
+    )
+  end
+
   def update_payment_amount!
     payment = current_order.payments.valid.amazon.first
     payment.amount = current_order.total
     payment.save!
   end
 
   def complete_amazon_order!
-    amazon_order = SpreeAmazon::Order.new(
```

Comment:
@acreilly Fair point, I think it reads nicer this way since it only uses data available to the whole controller but it definitely isn't required.  I have a vague recollection of being about to add another usage of amazon_order but I can't remember now, so I can pull it back in if you'd rather.